### PR TITLE
Debian packaging changes from libibverbs 1.2.1-2

### PR DESCRIPTION
**Pull request by bdrung**
A full review of the packaging differences between libibverbs 1.2.1-2 from Debian and the combined rdma-core package revealed some changes. These commits contain the changes that should be carried over to rdma-core.

### CODE REVIEW

File: debian/libibverbs-dev.links [hunk 1]
```diff
@@ -0,0 +1,16 @@
+usr/share/man/man3/ibv_alloc_pd.3		usr/share/man/man3/ibv_dealloc_pd.3
```
Comment (jgunthorpe) [inline debian/libibverbs-dev.links]: Why is the .links file necessary? The links are installed by cmake and dh does the right thing:

```
$ dpkg-deb -c ../libibverbs-dev_12-1_amd64.deb | grep -i alloc
-rw-r--r-- root/root       612 2017-01-31 09:27 ./usr/share/man/man3/ibv_alloc_pd.3.gz
-rw-r--r-- root/root       995 2017-01-31 09:27 ./usr/share/man/man3/ibv_alloc_mw.3.gz
lrwxrwxrwx root/root         0 2017-01-31 09:28 ./usr/share/man/man3/ibv_dealloc_mw.3.gz -> ibv_alloc_mw.3.gz
lrwxrwxrwx root/root         0 2017-01-31 09:28 ./usr/share/man/man3/ibv_dealloc_pd.3.gz -> ibv_alloc_pd.3.gz
```

If any are missing please update libibverbs/man/CMakeLists.

File: debian/rules [hunk 2]
```diff
@@ -1,5 +1,7 @@
 #!/usr/bin/make -f
 
+export DEB_BUILD_MAINT_OPTIONS=hardening=+all
+
 %:
```
Comment (jgunthorpe) [inline debian/rules]: I'm nervous about this - libibverbs is a performance library, adding security check overhead is probably not desired. Do you know how much overhead there is?

Comment (bdrung) [inline debian/rules]: No, I don't know how much performance impact it has. Can somebody can measuring that?

File: debian/libibverbs-dev.links (hunk 1 above)
Comment (bdrung) [inline debian/libibverbs-dev.links]: Yes, you are right. My bad. I checked that all symlinks are specified in libibverbs/man/CMakeLists. Thus I can drop that commit.

File: debian/rules (hunk 2 above)
Comment (jgunthorpe) [inline debian/rules]: Well, looking at the list:
- HARDENING_FORMAT, we are already compile with -Wformat=2, so this does nothing
- FORTIFY - This  probably does nothing on this tree, but is performance-neutral
- STACKPROTECTOR - this has a runtime cost
- PIE - We don't have any performance critical executables, and libraries are already PIC, so this is neutral - however we should check that this flag is passed around properly in cmake, it is a bit tricky..
- RELRO - No performance cost for us
- BINDNOW - No performance cost for us

I looked into it some more and FC25 is enabling -fstack-protector-strong by default now. As is Centos 7 - so I think it is safe for debian to do it as well. If someone determines it is a problem we will have to handle it globally...

Can you check that pie worked properly?

Comment (jgunthorpe) [inline debian/rules]: pie is not really passed into cmake properly, it is applying to shared libraries too.. But that seems to be OK. or at least if it isn't Debian will have to fix debhelper or the compiler...

**Pull request merged**